File: java/src/main/java/com/sopra/TPFinal/model/VideoProjecteur.java
package com.sopra.TPFinal.model;

import java.util.Set;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;

@Entity
@DiscriminatorValue("videoprojecteur")
public class VideoProjecteur extends Materiel {
	@OneToMany(fetch = FetchType.LAZY)
	private Set<Formation> formations;

	public VideoProjecteur() {
		super();
	}

	public Set<Formation> getFormations() {
		return formations;
	}

	public void setFormations(Set<Formation> formations) {
		this.formations = formations;
	}

}

File: angular/src/app/model/videoProjecteur.ts
import {Formation} from './Formation';
import {Materiel} from './materiel';

export class VideoProjecteur extends Materiel {

  constructor(private _formations?: Formation[]) {
    super();
    this.type = 'videoprojecteur';
  }

  get formationsVideoProj(): Formation[] {
    return this._formations;
  }

  set formationsVideoProj(value: Formation[]) {
    this._formations = value;
  }
}
